wgpu-core/Cargo.toml

```diff
@@ -29,6 +29,11 @@ targets = [
 # Cargo machete can't check build.rs dependencies. See https://github.com/bnjbvr/cargo-machete/issues/100
 ignored = ["cfg_aliases"]
 
+[lints.rust]
+# The `wgpu_validate_locks` config is supposed to be set via `RUSTFLAGS='--cfg wgpu_validate_locks'`
+# If set, `wgpu-core` uses the [`ranked`] module's locks. We hope to make this the default for debug builds soon.
+unexpected_cfgs = { level = "allow", check-cfg = ['cfg(wgpu_validate_locks)'] }
+
```

Comment:
thought: I've implemented something similar to this in [this commit](https://github.com/gfx-rs/wgpu/pull/6105/commits/fa898e4071f0d22857c7e7e909bbb3435c73d151) in #6105 as `build.rs` output instead:

```
println!("cargo::rustc-check-cfg=cfg(wgpu_validate_locks)");
```

The reason I did this was because I felt it made more sense to keep the `cfg` config (😆) in one place.

Reply:
🤷 no preference really, but I see you found one more than me.
So I can just cherry-pick your commit over